packages/parser/lib/css_parser.mly

```diff
@@ -89,29 +89,131 @@ bracket_block (X): xs = delimited(LEFT_BRACKET, X, RIGHT_BRACKET) { xs }
 paren_block (X): xs = delimited(LEFT_PAREN, X, RIGHT_PAREN) { xs }
 
 /* https://www.w3.org/TR/mediaqueries-5 */
-/* Parsing with this approach is almost as good the entire spec */
 
-prelude: xs = loption(nonempty_list(loc(value_in_prelude))) { xs }
-
-/* Missing grammars: */
-/* (width >= 600px) */
-/* (400px < width < 1000px) */
-/* (not (color)) and (not (hover)) */
-/* Combinator "," */
-media_query_prelude_item:
-  | i = IDENT { Ident i }
+// TODO: Alot of duplication here, any way to improve it?
+/* ">" */
+greater_operator: o = OPERATOR { o }
+/* "<" */
+less_operator: o = OPERATOR { o }
+/* "=" */
+equal_sign: e = OPERATOR { e }
+/* "and" */
+and_operator: o = OPERATOR { o }
+/* "or" */
+or_operator: o = OPERATOR { o }
+/* "only" */
+only_operator: o = OPERATOR { o }
+/* "not" */
+not: n = OPERATOR { n }
+/* "all" | "screen" | "print" */
+media_type: s = IDENT { s }
+
+/* https://www.w3.org/TR/mediaqueries-5/#mq-syntax */
+
+/* <mf-plain> = <mf-name> : <mf-value> */
+mf_plain: mf = IDENT WS? COLON WS? value { mf } 
+
+/* <mf-lt> = '<' '='? */
+mf_lt: mflt = less_operator equal_sign? { mflt }
+
+/* <mf-gt> = '>' '='? */
+mf_gt: mglt = greater_operator equal_sign? { mglt }
+
+/* <mf-comparison> = <mf-lt> | <mf-gt> | <mf-eq> */
+mf_comparison:
+  | mflt = mf_lt { mflt }
+  | mfgt = mf_gt { mfgt }
+  | mfeq = equal_sign { mfeq }
+
+/* <mf-range> =
+      <mf-name> <mf-comparison> <mf-value>
+      | <mf-value> <mf-comparison> <mf-name>
+      | <mf-value> <mf-lt> <mf-name> <mf-lt> <mf-value>
+      | <mf-value> <mf-gt> <mf-name> <mf-gt> <mf-value> */
+mf_range:
+  | xs = IDENT WS mf_comparison WS value { xs } 
+  | value WS xs =  mf_comparison WS IDENT { xs }
+  | value WS xs = mf_lt WS IDENT WS mf_lt WS value { xs } 
+  | value WS xs = mf_gt WS IDENT WS mf_gt WS value { xs } 
```

Comment:
The values could be interpolations right?

Do you want to add it in a further PR?

Reply:
> The values could be interpolations right?

Yes, I think I will add it to this PR!